tidy3d/components/data_new.py

```diff
@@ -0,0 +1,137 @@
+from typing import Dict, List, Union
+import json
+
+import pydantic as pd
+import h5py
+import xarray as xr
+
+from .base import Tidy3dBaseModel
+from .types import ArrayLike
+from .simulation import Simulation
+from .monitor import FieldMonitor, FieldTimeMonitor, FluxMonitor, FluxTimeMonitor, ModeMonitor
+from .monitor import PermittivityMonitor, ModeFieldMonitor
+
+# every type that can be used as a component
+ComponentType = Union[
+    Simulation,
+    FieldMonitor,
+    FieldTimeMonitor,
+    FluxMonitor,
+    FluxTimeMonitor,
+    ModeMonitor,
+    ModeFieldMonitor,
+]
```

Comment:
I was hoping to use `Tidy3dData` also for the `ModeSolverData`, which is defined in the mode solver plugin right now. Perhaps either the `ModeSolverData` or the whole mode solver could be moved to the core tidy3d. But more generally do we want some way for plugins to define their own Tidy3dData types, and what happens if the component is not in the core components?

Reply:
Hm, so you were thinking to put the `ModeSolver` as the component here?  Yea I originally had it accept `Tidy3dBaseModel` as the type, but realized there might be some ambiguity when loading this from file.  Fundamentally, we'll need to know either what the original `type` of `component` was, or a list of accepted `types` to try to load it as.  We can always leave the `component` blank as well.

Reply:
Yeah, `ModeSolver` would be the component.